client/src/components/Panels/Common/ToolAdvancedSearchResults.vue

```diff
@@ -107,14 +107,17 @@ export default {
         };
     },
     created() {
-        const initData = { summary: "", help_text: "", _showDetails: false, is_fetched: false };
-        this.buffer = this.tools.slice(0, this.bufferLen);
-        this.buffer.forEach((tool) => Object.assign(tool, { ...initData }));
-        this.buffer.forEach(async (tool, index) => {
-            await this.fetchHelp(index);
-        });
+        this.initialize();
     },
     methods: {
+        initialize() {
+            if (this.tools) {
+                this.tools.slice(this.bufferLen - defaultBufferLen, this.bufferLen).forEach(async (toolId) => {
+                    await this.fetchHelp(toolId);
+                });
+                this.busy = false;
+            }
+        },
```

Comment:
Just fyi, @ahmedhamidawan and I talked offline and we will revise the data processing mechanism, which should significantly improve the overall performance.